app/src/main/java/org/jellyfin/androidtv/util/RecommendedServerIssueExtensions.kt

```diff
@@ -0,0 +1,59 @@
+package org.jellyfin.androidtv.util
+
+import android.content.Context
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.auth.ServerRepository
+import org.jellyfin.sdk.api.client.exception.ApiClientException
+import org.jellyfin.sdk.discovery.RecommendedServerIssue
+import javax.net.ssl.SSLHandshakeException
+
+private fun ApiClientException.getFriendlyMessage(context: Context): String? = when (cause) {
+	is SSLHandshakeException -> context.getString(R.string.server_issue_ssl_handshake)
+	else -> null
+}
+
+/**
+ * Create a localized string with the message to show an end-user of the app.
+ */
+fun RecommendedServerIssue.getFriendlyMessage(context: Context): String = when (this) {
+	is RecommendedServerIssue.MissingSystemInfo -> throwable?.let { throwable ->
+		// Search for exceptions we want to display but the SDK does not have
+		// an RecommendedServerIssue member for
+		if (throwable is ApiClientException) throwable.getFriendlyMessage(context)
+		else null
+	} ?: context.getString(R.string.server_issue_unable_to_connect)
+
+	is RecommendedServerIssue.InvalidProductName -> context.getString(R.string.server_issue_invalid_product)
+
+	RecommendedServerIssue.MissingVersion -> context.getString(R.string.server_issue_missing_version)
+
+	is RecommendedServerIssue.UnsupportedServerVersion -> context.getString(
+		R.string.server_issue_unsupported_version,
+		version.toString(),
+	)
+
+	is RecommendedServerIssue.OutdatedServerVersion -> context.getString(
+		R.string.server_issue_outdated_version,
+		version.toString(),
+		ServerRepository.minimumServerVersion.toString()
+	)
+
+	is RecommendedServerIssue.SlowResponse -> context.getString(R.string.server_issue_timeout)
```

Comment:
It's unfortunate that the name for this property is SlowResponse and not Timeout. I see that's defined by the SDK though.

Is it safe to assume that SlowResponse always means timeout, or could SlowResponse be caught in other situations?

I see "slow http response" in my server logs all the time. I assume that's unrelated to anything the "discovery" part of the SDK you're using here would test for.

Reply:
SlowResponse is literally "it took to long for a proper reply so we gave up trying this server". It's sort of a timeout but it doesn't use the timeout value from JellyfinOptions. For discovery the value is 3 seconds [^1] because the response is not difficult to calculate and if this response is 3+ seconds, others will definitely be more making a client unusable.

A timeout in the ApiClient is reported as MissingSystemInfo with the [TimeoutException](https://github.com/jellyfin/jellyfin-sdk-kotlin/blob/master/jellyfin-api/src/commonMain/kotlin/org/jellyfin/sdk/api/client/exception/TimeoutException.kt) as cause.

[^1]: https://github.com/jellyfin/jellyfin-sdk-kotlin/blob/master/jellyfin-core/src/commonMain/kotlin/org/jellyfin/sdk/discovery/RecommendedServerDiscovery.kt#L29

Reply:
The "slow http response" in server logs is something calculated by the server and unrelated to the SDK/app(s).

Reply:
When I put in the IP address of a computer that is on my LAN but isn't running a jellyfin server I get this:
`D/RecommendedServerDiscovery: Could not connect to http://192.168.3.200:8096
    org.jellyfin.sdk.api.client.exception.TimeoutException: HTTP request timed out`

And the feedback is all "Unable to connect". Is that expected behavior?

Otherwise everything looks good!
I tested with the demo server, my server, and the badssl example

Reply:
>  Is that expected behavior (unable to connect instead of timed out)?

Yes because it couldn't connect because of a timeout